src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1733,18 +1738,71 @@ subroutine li_apply_front_ablation_velocity(meshPool, geometryPool, velocityPool
             uvelSum = 0.0_RKIND
             vvelSum = 0.0_RKIND
             thkCount = 0
+            nGroundedNeighbors = 0
             do iNeighbor = 1, nEdgesOnCell(iCell)
                jCell = cellsOnCell(iNeighbor, iCell)
-               if (li_mask_is_dynamic_ice(cellMask(jCell))) then
+               if ( li_mask_is_dynamic_ice(cellMask(jCell)) .and. li_mask_is_floating_ice(cellMask(jCell)) ) then
                   thkSum = thkSum + thicknessForAblation(jCell)
                   uvelSum = uvelSum + uvelForAblation(jCell)
                   vvelSum = vvelSum + vvelforAblation(jCell)
                   ablationVelSum = ablationVelSum + ablationVelocity(jCell)
                   thkCount = thkCount + 1
+               else if ( li_mask_is_grounding_line(cellMask(jCell)) .or. li_mask_is_grounded_ice(cellMask(jCell)) ) then
+                  nGroundedNeighbors = nGroundedNeighbors + 1
                endif
             enddo
+
+            ! if floating non-dynamic cell has any grounded neighbors, use ice
```

Comment:
The current logic ignores the case where the cell being evaluated has a mix of floating dynamic neighbors and grounded neighbors.  In that case, the code will ignore the floating dynamic neighbor and only evaluate the grounded neighbors two-back.  Is that what we want?  What I'm describing seems like an edge case.  It may occur at the lateral margin of an ice shelf in an embayment.

Reply:
The current logic does not in fact ignore that case. That section is called if the non-dynamic cell has one grounded (and/or grounding line) neighbor, but then there is no logic about grounded ice in the actual implementation.
oneCellBackList is defined as any cell with ice, then twoCellsBackList is defined as any dynamic cell that is not is oneCellBackLIst (or already in twoCellsBackList).